src/common/dataStructure/trie/trie.ts

```diff
@@ -1,52 +1,51 @@
+/* eslint-disable @typescript-eslint/no-this-alias */
 import {
-    IsPresentInTrie,
-    TrieLookupStatus,
-    IsNotPresentInTrie,
-    IsPartiallyPresentInTrie
-} from "./types/trieLookupStatus.js";
+  IsPresentInTrie,
+  TrieLookupStatus,
+  IsNotPresentInTrie,
+  IsPartiallyPresentInTrie,
+} from './types/trieLookupStatus.js';
 
 export class Trie {
+  static readonly isPresentInTrie: IsPresentInTrie = 2;
+  static readonly isPartiallyPresentInTrie: IsPartiallyPresentInTrie = 1;
+  static readonly isNotPresentInTrie: IsNotPresentInTrie = 0;
 
-    static readonly isPresentInTrie: IsPresentInTrie = 2;
-    static readonly isPartiallyPresentInTrie: IsPartiallyPresentInTrie = 1;
-    static readonly isNotPresentInTrie: IsNotPresentInTrie = 0;
+  static readonly dummyKeyMarkingPresenceInTrie = '0';
 
-    static readonly dummyKeyMarkingPresenceInTrie = "0";
+  [property: string]: Trie | boolean | Function;
 
-    [property: string]: Trie | boolean | Function;
-
-    constructor(keywords: Array<string>) {
-        for (const keyword of keywords) {
-            let current: Trie = this;
-            for (const char of keyword) {
-                if (!(char in current)) {
-                    current[char] = new Trie([]);
-                }
-                current = current[char] as Trie;
-            }
-            current[Trie.dummyKeyMarkingPresenceInTrie] = true;
+  constructor(keywords: Array<string>, trie?: Trie) {
```

Comment:
We should update the Trie class to have a private field that would encapsulate the trie for a given set of keywords. I assume that passing another trie object in the factory method was supposed to perform cloning of a trie / updating existing trie. We should extract these things into "clone" and "static fromTrie" functions

Reply:
I temporarily removed the second argument as it has no value for MVP and should not be directly ported